docs/design/modular_installation.md

```diff
@@ -0,0 +1,63 @@
+# Kuadrant Proposal - Modular Installation
+
+Kuadrant is developing a set of loosely coupled functionalities built directly on top of Kubernetes.
+Kuadrant aims to allow customers to just install, use and understand those functionalities they need.
+Currently, the installation tool of kuadrant, the [kuadrantctl CLI](https://github.com/Kuadrant/kuadrantctl),
+installs all or nothing. For instance, a customer not wanting to rate limit the API,
+should not have to install the rate limit component ([Limitador](https://github.com/Kuadrant/limitador)).
+
+## High Level Goals
+
+* Install only required components. Operate only required components.
+* Expose only the kuadrant API that has available implementation by the with the deployed components
+
+## Installation profiles
+
+The kuadrant installation mechanism should offer modular installation to enable pieces of kuadrant.
+Modular installation options should be feature oriented rather than deployment component oriented.
+Then, it is up to the installation tool to decide what components need to be deployed and how to
+configure it.
+
+Kuadrant features:
+
+* Multiple APIs bundled in a product
+* Service Discovery
+* TLS
+* Authentication (mTLS, API key, OIDC)
+* Rate Limiting (pre-auth, post-auth)
+
+Each feature, or part of it, is eligible to be included or excluded when installing kuadrant.
+
+Some profiles can be defined to group set of commonly required features. Naming the profiles
+allows the customer to easily express wanted installation configuration. Furthermore, profiles
+not only can be used to group a set of features, profiles can be used to define deployment options.
+
+| Name | Description |
+| ---  | --- |
+| **Minimal** | Minimal installation required to run an API without any protection, analytics or API management |
+| **Authenticated** | MTLS, API key and OIDC authentication features |
+| **RateLimit** | Basic rate limit (only pre-auth rate limit) features |
+| **Full** | Full featured kuadrant installation |
+| **Development** | Full featured kuadrant for development purposes. Development deployment mode: debug log level, no cache, tracing |
```

Comment:
Does a development profile make sense? Seems different from the intention of the other profiles, and likely something you'd want to enabled on any configuration Minimum, Full etc..., or, is the intention that you can stack these profiles, have more than one enabled? In which case is this just a list of the features you can enable?

Reply:
Agreed. The flow for developers might be different, but it's a good thing that they interact with the system under the same deployment options and capabilities (same profile?) we offer to users.

As an example, one who's working on a feature to which the _RateLimit_ profile is supposed to suffice should not inadvertently count on features available on the _Auth_ profile, or else an automatic and perhaps unexpected requirement for using the _Full_ profile could emerge. At the very least, the boundaries between one profile and the other should be clear not only to users, but to developers as well and possibly most of all, IMHO.

That said, I'm guessing the intent for a _Development_ profile was to provide a "shortcut" to activate different log mode, disable caching, etc. Maybe the CLI can still offer that as a `--development` flag that can be used in combination with all the other profiles instead of alternatively to those?

Reply:
I was inspired by the [istio installation profiles](https://istio.io/latest/docs/setup/additional-setup/config-profiles/). There is this `demo` profile that is what I meant by `Development`. Istio profiles are more complex and span a wider set of uses cases. For instance, you can have different profiles for the control plane and the data plane. Which, of course, can make sense here too.

I will re-think this again

Reply:
ok, removed the "Development" deployment option. That is a property that could be activated in any of the deployment options available.